### DISCUSSION TITLE
Sim keeps blowing up.. tried many things

### DISCUSSION THREAD

**Discussion by GaN-T** [NanoComp/meep#2384] [Q&A]
(EDITED: Please ignore the previous edits of this post, the issue posted below is a simpler example of the problem I am having)

I am trying a simple simulation to visualise a gaussian pulse scattering off and around a thin metal strip. I am following the example shown on the [Simpetus project page no.  4](http://www.simpetus.com/projects.html#meep_image_sensor). In my case I am using the same materials and similar simulation object but the simulation fields blow up and the run never completes. I get the following message when the process terminates.
```

meep: simulation fields are NaN or Inf
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 2

```

Does anyone know how to set this up correctly?

the cell viewed in YZ crosssection:

![image](https://user-images.githubusercontent.com/120845037/214879265-79b9066f-9b11-46f4-89da-44e6ee5d2bb2.png)

```
import meep as mp
import numpy as np
from datetime import datetime
from meep.materials import W

import matplotlib.pyplot as plt

um_scale = 1.0 
resolution = 30
eV_um_scale = um_scale/1.23984193

cell_x = 8  # cellsize in x directions
cell_y = 8 # cellsize in x directions
cell_z = 10 # cellsize in x directions
cell= mp.Vector3(cell_x,cell_y,cell_z)

W_t = 0.2   # z
W_l = 1  # y
W_w = 1   # x
pad = 0.5

geometry = [
            mp.Block(mp.Vector3(W_w , W_l, W_t),
            center = mp.Vector3(0,0,W_t/2),
            material = W)]

f_min = 1
f_max  = 1/0.7
fcen = 0.5*(f_max + f_min)
df = f_max - f_min     # <= pulse width in frequency.    

  # half of max wavelength in sim
dpml = 0.5*1/f_min

pml_layers = [mp.PML(dpml, direction=mp.Z, side=mp.High),
                     mp.Absorber(dpml, direction=mp.Z, side=mp.Low),
                     ]

src_zpos = cell_z/2-dpml-pad

sources = [mp.Source(mp.GaussianSource(fcen, fwidth=df),
                        component=mp.Ey,
                        center=mp.Vector3(0,0,src_zpos),
                        size=mp.Vector3(cell_x,cell_y))]

sim = mp.Simulation(cell_size= cell, 
                    boundary_layers = pml_layers,
                    geometry = geometry,
                    sources = sources,
                    dimensions=3,
                    resolution = resolution,
                    split_chunks_evenly=False,
                    k_point=mp.Vector3())


pt = mp.Vector3(0,0,-cell_z/2 + dpml + 2)
print("STARTING AT:", datetime.now())
sim.run(mp.to_appended("ey", mp.at_every(3, mp.output_efield_y)),
        until_after_sources=mp.stop_when_fields_decayed(10, mp.Ey, pt, 1e-3))





```

**Comment by oskooi** [comment]
See the following FAQ for a list of things to try: https://meep.readthedocs.io/en/latest/FAQ/#why-are-the-fields-blowing-up-in-my-simulation.

**Comment by GaN-T** [comment]
Thank you for your reply. I think I found the culprit. There were two problems. 

I was using the wrong python version with mpirun to run the script. In my parallel meep conda env calling python3 and python are different on my system (just used anaconda defaults). This was causing the meep to eventually crash at the step where it begins to write the fields to the file. Simply correcting the mpirun command to run the script using mpirun using the correct python solved this first issue. 

Next was the material definition, when I use Tungsten as imported from the meep materials library my script doesnt work. However, using the material definition from Example 4 on the simpetus project page, the simulation runs fine and I can visualise the output.

Finally can see the result !! :D
![strip](https://user-images.githubusercontent.com/120845037/215236277-0b8b0346-a142-475f-93c1-973c980b0d85.gif)


 Does anyone know why the standard W causes this script to end incorrectly? The meep materials library definition of Tungsten has 2 additional LorentzianSusceptibility terms compared to the one used in the Simpetus project example 4. The values are otherwise identical.
